layer1/layer1.yaml

```diff
@@ -10,124 +10,112 @@ components:
+            Enable promiscuous mode if supported.
           type: boolean
           default: false
-        link_training:
+        mtu:
           description: >-
-            Enable/disable link training.
-          type: boolean
-          default: false
-        rs_fec:
+            Set the maximum transmission unit size if supported.
+          type: integer
+          default: 1500
+        ieee_media_defaults:
           description: >-
-            Enable/disable reed solomon forward error correction (RS FEC).
+            Set to true to override the auto_negotiate, link_training
+            and rs_fec settings for gigabit ethernet interfaces.
           type: boolean
-          default: false
-        speed:
+          default: true
+        auto_negotiate:
           description: >-
-            This is the speed that will be used if auto_negotiate is false.
-          type: string
-          enum: [one_hundred_gbps, fifty_gbps, forty_gbps, twenty_five_gpbs, ten_gbps]
-          default: one_hundred_gbps
+            Enable/disable auto negotiation.
+          type: boolean
+          default: true
+        auto_negotiation:
+          $ref: '#/components/schemas/Layer1.AutoNegotiation'
         flow_control:
           $ref: '#/components/schemas/Layer1.FlowControl'
-    
-    Layer1.EthernetVirtual:
-      description: >-
-        Container for virtual ethernet settings
-      type: object
-      properties:
-        speed:
-          description: >-
-            This is the speed that will be set on the virtual ethernet interface.
-          type: string
-          enum: [one_hundred_mbps, one_gbps, ten_gbps, twenty_five_gbps, fifty_gbps, one_hundred_gbps]
-          default: ten_gbps
 
-    Layer1.Ethernet:
+    Layer1.AutoNegotiation:
       description: >-
-        Container for 10/100/1000 ethernet settings
+        Container for auto negotiation settings
       type: object
       properties:
-        media:
-          type: string
-          enum: [copper, fiber]
-          default: copper
-        speed:
-          description: >-
-            This is the speed that will be used if auto_negotiate is false.
-          type: string
-          enum: [one_thousand_mbps, one_hundred_fd_mbps, one_hundred_hd_mbps, ten_fd_mbps, ten_hd_mbps]
-          default: one_thousand_mbps
-        auto_negotiate:
+        advertise_1000_mbps:
           description: >-
-            Enable/disable auto negotiation.
+            If auto_negotiate is true and the interface supports this option
+            then this speed will be advertised.
           type: boolean
           default: true
-        advertise_one_thousand_mbps:
+        advertise_100_fd_mbps:
           description: >-
-            If auto_negotiate is true then this speed will be advertised.
+            If auto_negotiate is true and the interface supports this option
+            then this speed will be advertised.
           type: boolean
           default: true
```

Comment:
Two queries:
- Shouldn't all `advertise_*` default to false ? i.e. doesn't make sense to have advertise_100_fd_mbps and advertise_100_hd_mbps set to true at the same time. They also are not needed when rs_fec=True (since it applies to gigabit eth).
- Is FD (full duplex) and HD (half duplex) documented somewhere ?